README.rst
=============================
pympa-registrum
=============================

.. image:: https://badge.fury.io/py/pympa-registrum.png
    :target: https://badge.fury.io/py/pympa-registrum

.. image:: https://travis-ci.org/simodalla/pympa-registrum.png?branch=master
    :target: https://travis-ci.org/simodalla/pympa-registrum

.. image:: https://coveralls.io/repos/simodalla/pympa-registrum/badge.png?branch=master
    :target: https://coveralls.io/r/simodalla/pympa-registrum?branch=master

Your project description goes here

Documentation
-------------

The full documentation is at https://pympa-registrum.readthedocs.org.

Quickstart
----------

Install pympa-registrum::

    pip install pympa-registrum

Then use it in a project::

    import pympa-registrum

Features
--------

* TODO


HISTORY.rst
.. :changelog:

History
-------

0.1.0 (2014-11-26)
++++++++++++++++++

* First release on PyPI.


tests/test_models.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

"""
test_pympa-registrum
------------

Tests for `pympa-registrum` models module.
"""

import os
import shutil
import unittest

from pympa_registrum import models


class TestPympa_registrum(unittest.TestCase):

    def setUp(self):
        pass

    def test_something(self):
        pass

    def tearDown(self):
        pass


runtests.py
import sys

try:
    from django.conf import settings

    settings.configure(
        DEBUG=True,
        USE_TZ=True,
        DATABASES={
            "default": {
                "ENGINE": "django.db.backends.sqlite3",
            }
        },
        ROOT_URLCONF="pympa_registrum.urls",
        INSTALLED_APPS=[
            "django.contrib.auth",
            "django.contrib.contenttypes",
            "django.contrib.sites",
            "pympa_registrum",
        ],
        SITE_ID=1,
        NOSE_ARGS=['-s'],
    )

    try:
        import django
        setup = django.setup
    except AttributeError:
        pass
    else:
        setup()

    from django_nose import NoseTestSuiteRunner
except ImportError:
    import traceback
    traceback.print_exc()
    raise ImportError("To fix this error, run: pip install -r requirements-test.txt")


def run_tests(*test_args):
    if not test_args:
        test_args = ['tests']

    # Run tests
    test_runner = NoseTestSuiteRunner(verbosity=1)

    failures = test_runner.run_tests(test_args)

    if failures:
        sys.exit(failures)


if __name__ == '__main__':
    run_tests(*sys.argv[1:])


pympa_registrum/admin.py
# -*- coding: utf-8 -*-
from __future__ import unicode_literals, absolute_import

from django.contrib import admin

from .models import Comunicazione, MezzoComunicazione


class ComunicazioneAdmin(admin.ModelAdmin):
    actions = ['evadi']
    date_hierarchy = 'ricezione_data'
    fields = ['ricezione_data', 'destinatari', 'mittente_cognome',
              'mittente_nome', 'mezzo', 'motivazione', 'recapito', 'note',
              'evasa']
    filter_horizontal = ['destinatari']
    list_display = ['id', 'ricezione_data', 'ld_elenco_destinatari', 'evasa',
                    'mittente_cognome', 'mittente_nome',
                    'recapito', 'motivazione', 'ld_ricevente']
    list_filter = ['evasa', 'ricezione_data', 'destinatari', 'ricevente',
                   'mezzo']
    list_per_page = 30
    list_select_related = True
    ordering = ['-id', 'mittente_cognome', 'mittente_nome']
    radio_fields = {'mezzo': admin.VERTICAL}
    search_fields = ['destinatari__last_name', 'destinatari__first_name',
                     'mittente_cognome', 'mittente_nome']

    def evadi(self, request, queryset):
        rows_updated = queryset.update(evasa=True)
        if rows_updated == 1:
            message_bit = "Comunicazione evasa correttamente"
        else:
            message_bit = "Comunicazioni selezionate evase correttamente"
        self.message_user(request, message_bit)
    evadi.short_description = "Evadi la comunicazione"

    def ld_elenco_destinatari(self, obj):
        destinatari = ['<li>{user.last_name} {user.first_name}</li>'.format(
            user=d) for d in obj.destinatari.all()]
        return "<ul>{}</ul>".format(''.join(destinatari))
    ld_elenco_destinatari.short_description = 'Destinatari'
    ld_elenco_destinatari.allow_tags = True

    def ld_ricevente(self, obj):
        return "{user.last_name} {user.first_name}".format(user=obj.ricevente)
    ld_ricevente.short_description = 'Ricevente'
    ld_ricevente.admin_order_field = 'ricevente'

    def save_model(self, request, obj, form, change):
        obj.ricevente = request.user
        obj.save()


admin.site.register(MezzoComunicazione)
admin.site.register(Comunicazione, ComunicazioneAdmin)

pympa_registrum/models.py
# -*- coding: utf-8 -*-
from __future__ import unicode_literals

from django.conf import settings
from django.db import models
from django.utils.encoding import python_2_unicode_compatible


# User = get_user_model()


@python_2_unicode_compatible
class MezzoComunicazione(models.Model):
    nome = models.CharField(max_length=30)
    desc = models.TextField('Descrizione', blank=True, null=True)

    def __str__(self):
        return self.nome

    class Meta:
        verbose_name_plural = "Mezzi di Comunicazione"


@python_2_unicode_compatible
class Comunicazione(models.Model):
    ricezione_data = models.DateTimeField(
        verbose_name='Data e ora di ricezione',
        help_text='Seleziona ora e data di ricezione della comunicazione')
    mezzo = models.ForeignKey(MezzoComunicazione, verbose_name='Ricevuta con',
                              help_text="Seleziona il mezzo con cui e' stata"
                                        " ricevuta la comunicazione")
    destinatari = models.ManyToManyField(
        settings.AUTH_USER_MODEL, blank=True, null=True, related_name="destinatari",
        limit_choices_to={'groups__name__exact': 'registrum_destinatari'})
    mittente_cognome = models.CharField(blank=True, null=True, max_length=200,
                                        verbose_name='Cognome')
    mittente_nome = models.CharField(blank=True, null=True, max_length=200,
                                     verbose_name='Nome')
    motivazione = models.TextField(blank=True, null=True)
    recapito = models.CharField(blank=True, null=True, max_length=300,
                                help_text='Recapito del mittente')
    note = models.TextField(blank=True, null=True)
    ricevente = models.ForeignKey(
        settings.AUTH_USER_MODEL, blank=True, null=True, related_name="ricevente",
        limit_choices_to={'groups__name__exact': 'registrum_riceventi'})
    evasa = models.BooleanField(default=False)

    def __str__(self):
        return 'comunicazione n. {}'.format(self.id)

    class Meta:
        verbose_name_plural = "Comunicazioni"


pympa_registrum/migrations/0001_initial.py
# -*- coding: utf-8 -*-
from __future__ import unicode_literals

from django.db import models, migrations
from django.conf import settings


class Migration(migrations.Migration):

    dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.CreateModel(
            name='Comunicazione',
            fields=[
                ('id', models.AutoField(verbose_name='ID', primary_key=True, serialize=False, auto_created=True)),
                ('ricezione_data', models.DateTimeField(verbose_name='Data e ora di ricezione', help_text='Seleziona ora e data di ricezione della comunicazione')),
                ('mittente_cognome', models.CharField(blank=True, verbose_name='Cognome', null=True, max_length=200)),
                ('mittente_nome', models.CharField(blank=True, verbose_name='Nome', null=True, max_length=200)),
                ('motivazione', models.TextField(blank=True, null=True)),
                ('recapito', models.CharField(blank=True, max_length=300, null=True, help_text='Recapito del mittente')),
                ('note', models.TextField(blank=True, null=True)),
                ('evasa', models.BooleanField(default=False)),
                ('destinatari', models.ManyToManyField(blank=True, related_name='destinatari', to=settings.AUTH_USER_MODEL, null=True)),
            ],
            options={
                'verbose_name_plural': 'Comunicazioni',
            },
            bases=(models.Model,),
        ),
        migrations.CreateModel(
            name='MezzoComunicazione',
            fields=[
                ('id', models.AutoField(verbose_name='ID', primary_key=True, serialize=False, auto_created=True)),
                ('nome', models.CharField(max_length=30)),
                ('desc', models.TextField(blank=True, verbose_name='Descrizione', null=True)),
            ],
            options={
                'verbose_name_plural': 'Mezzi di Comunicazione',
            },
            bases=(models.Model,),
        ),
        migrations.AddField(
            model_name='comunicazione',
            name='mezzo',
            field=models.ForeignKey(to='pympa_registrum.MezzoComunicazione', help_text="Seleziona il mezzo con cui e' stata ricevuta la comunicazione", verbose_name='Ricevuta con'),
            preserve_default=True,
        ),
        migrations.AddField(
            model_name='comunicazione',
            name='ricevente',
            field=models.ForeignKey(blank=True, to=settings.AUTH_USER_MODEL, related_name='ricevente', null=True),
            preserve_default=True,
        ),
    ]
